projects/hip-tests/catch/perftests/graph/hipGraphTopology.cc

```diff
@@ -0,0 +1,462 @@
+        }
+      }
+    }
+  } else if (opt.topology == "mixed") {
+    // Pattern: memset -> 3 kernels -> memcpy -> 2 kernels -> empty -> 3 kernels -> memset
+    std::vector<hipGraphNode_t> all_nodes;
+    std::vector<std::string> node_types;
+    std::vector<int> kernel_batches;
+    hipGraphNode_t prev{};
+    
+    int kernel_count = 0;
+    int current_batch_size = 0;
+    int batch_number = 0;
+    
+    for (int i = 0; i < opt.length; ++i) {
+      std::string node_type;
+      if (i == 0) {
+        // Start with memset
+        prev = add_memset_node(graph, nullptr, 0, d_mem1, 0, mem_size);
+        all_nodes.push_back(prev);
+        node_type = "memset";
+        if (current_batch_size > 0) {
+          kernel_batches.push_back(current_batch_size);
+          current_batch_size = 0;
+          batch_number++;
+        }
+      } else {
+        int step = i % 9; // 9-step pattern
+        if (step == 1 || step == 2 || step == 3) {
+          // 3 consecutive kernels
+          hipGraphNode_t n = add_kernel_node(graph, &prev, 1, opt.iters);
+          all_nodes.push_back(n);
+          prev = n;
+          node_type = "kernel";
+          kernel_count++;
+          current_batch_size++;
+        } else if (step == 4) {
+          // memcpy (breaks batching)
+          if (current_batch_size > 0) {
+            kernel_batches.push_back(current_batch_size);
+            current_batch_size = 0;
+            batch_number++;
+          }
+          hipGraphNode_t n = add_memcpy_node(graph, &prev, 1, d_mem2, d_mem1, mem_size);
+          all_nodes.push_back(n);
+          prev = n;
+          node_type = "memcpy";
+        } else if (step == 5 || step == 6) {
+          // 2 consecutive kernels
+          hipGraphNode_t n = add_kernel_node(graph, &prev, 1, opt.iters);
+          all_nodes.push_back(n);
+          prev = n;
+          node_type = "kernel";
+          kernel_count++;
+          current_batch_size++;
+        } else if (step == 7) {
+          // empty node
+          if (current_batch_size > 0) {
+            kernel_batches.push_back(current_batch_size);
+            current_batch_size = 0;
+            batch_number++;
+          }
+          hipGraphNode_t n = add_empty_node(graph, &prev, 1);
+          all_nodes.push_back(n);
+          prev = n;
+          node_type = "empty";
+        } else if (step == 8 || step == 0) {
+          // kernel nodes
+          hipGraphNode_t n = add_kernel_node(graph, &prev, 1, opt.iters);
+          all_nodes.push_back(n);
+          prev = n;
+          node_type = "kernel";
+          kernel_count++;
+          current_batch_size++;
+        }
+      }
+      node_types.push_back(node_type);
+      printf("Node %3d: %-8s", i, node_type.c_str());
+      if (i % 9 == 8 || i == opt.length - 1) printf("\n");
+      else printf(" -> ");
+    }
+    
+    if (current_batch_size > 0) {
+      kernel_batches.push_back(current_batch_size);
+    }
+    
+    printf("\nMixed topology summary:\n");
```

Comment:
Please use method "CONSOLE_PRINT" which is in one of the headers rather than directly usign printf here.